lib/octopus_serverspec_extensions/type/octopus_deploy_worker.rb

```diff
@@ -0,0 +1,173 @@
+      end
+
+      if (exists?)
+        thumbprint = `"c:\\program files\\Octopus Deploy\\Tentacle\\Tentacle.exe" show-thumbprint --console --nologo --instance #{instance}`
+        thumbprint = thumbprint.gsub('==== ShowThumbprintCommand starting ====', '').strip
+        thumbprint = thumbprint.gsub('The thumbprint of this Tentacle is: ', '').strip
+        thumbprint = thumbprint.gsub('==== ShowThumbprintCommand completed ====', '').strip
+        thumbprint = thumbprint.gsub('==== ShowThumbprintCommand ====', '').strip
+
+        @serverSupportsSpaces = check_supports_spaces(serverUrl)
+
+        if (@serverSupportsSpaces)
+          @spaceFragment = "#{@spaceId}/"
+        end
+
+        @worker = get_worker_via_api(serverUrl, apiKey, thumbprint)
+      else
+        raise "tentacle.exe does not exist"
+      end
+    end
+
+    def registered_with_the_server?
+      !@worker.nil?
+    end
+
+    def online?
+      return nil if @worker.nil?
+      @worker = poll_until_worker_has_completed_healthcheck(@serverUrl, @apiKey, @worker["Thumbprint"])
+      status = @worker['Status']
+      if ("#{status}" == "")
+        status = @worker['HealthStatus'] if "#{status}" == ""
+        puts "Expected status 'Healthy|HasWarnings' for Worker #{@name}, but got '#{status}'" if (status != "Healthy" && status != "HasWarnings")
+        status == "Healthy" || status == "HasWarnings"
+      else
+        puts "Expected status 'Online|CalamariNeedsUpgrade|NeedsUpgrade' for Worker #{@name}, but got '#{status}'" if (status != "Online" && status != "CalamariNeedsUpgrade" && status != "NeedsUpgrade")
+        status == "Online" || status == "CalamariNeedsUpgrade" || status == "NeedsUpgrade"
+      end
+    end
+
+    def in_space?(space_name)
+      return false if @worker.nil?
+      return false if @serverSupportsSpaces
+      url = "#{@serverUrl}/api/spaces/all?api-key=#{@apiKey}"
+      resp = Net::HTTP.get_response(URI.parse(url))
+      spaces = JSON.parse(resp.body)
+      space_id = spaces.select {|e| e["Name"] == space_name}.first["Id"]
+      @worker["SpaceId"] == space_id
+    end
+
+    def has_policy?(policy_name)
+      return false if @worker.nil?
+      url = "#{@serverUrl}/api/#{@spaceFragment}machinepolicies/all?api-key=#{@apiKey}"
+      resp = Net::HTTP.get_response(URI.parse(url))
+      policies = JSON.parse(resp.body)
+      policy_id = policies.select {|e| e["Name"] == policy_name}.first["Id"]
+      @worker["MachinePolicyId"] == policy_id
+    end
+
+    def has_display_name?(name)
+      return false if @worker.nil?
+      @worker["Name"] == name
+    end
+
+    def has_endpoint?(uri)
+      return false if @worker.nil?
+      return false if @worker["Uri"].nil? # polling tentacles have null endpoint. catch that.
+      puts "Expected uri '#{uri}' for Worker #{@name}, but got '#{@worker["Uri"]}'" unless (@worker["Uri"].casecmp(uri) == 0)
+      @worker["Uri"].casecmp(uri) == 0
+    end
+
+    def listening_tentacle?
```

Comment:
`listening_worker`?

Reply:
My thought was keep it the same as tentacle, so that if/when we eventually port to an inheritance-based model, there'll be less messing about